src/fmllauncher/java/net/minecraftforge/fml/loading/moddiscovery/ModsFolderLocator.java

```diff
@@ -58,7 +58,7 @@ public ModsFolderLocator() {
                 sorted(Comparator.comparing(path-> StringUtils.toLowerCase(path.getFileName().toString()))).
                 filter(p->StringUtils.toLowerCase(p.getFileName().toString()).endsWith(SUFFIX)).
                 map(p->new ModFile(p, this)).
-                peek(f->modJars.compute(f, (mf, fs)->createFileSystem(mf))).
+                filter(f->modJars.compute(f, (mf, fs)->createFileSystem(mf))!=null).
```

Comment:
What issue is this fixing? Where is it failing?

Reply:
It removes `null`s in the return value of the `scanMods` method.

Reply:
But that return value should *never* be null. So this is a bandaid and not a fix. Regardless it does not belong in this PR, keep separate things separated.

Reply:
This fails when you put an invalid jarfile into the mods folder. createFileSystem then returns null (after warning in the log that it's ignoring an invalid jarfile) so null (or rather nothing) ends up in modJars, but the mod is not actually disabled, leading to errors when it tries to read from the null FileSystem.

Try it out yourself by just adding an empty text file called "test.jar" to the mods directory.

Reply:
Please keep PRs tightly scoped to a single issue.